Tests/SecureXPCTests/Client & Server/Server Termination Integration Test.swift

```diff
@@ -0,0 +1,72 @@
+    func testShutdownServer() throws {
+        // Server & client setup
+        let echoRoute = XPCRouteWithMessageWithReply("echo", messageType: String.self, replyType: String.self)
+        let server = XPCServer.makeAnonymousService(clientRequirements: dummyRequirements)
+        try server.registerRoute(echoRoute) { msg in
+            return "echo: \(msg)"
+        }
+        server.start()
+        let client = XPCClient.forEndpoint(server.endpoint)
+        
+        // Send a message, which will result in the connection being established with the server
+        try client.sendMessage("1st message", route: echoRoute) { _ in }
+        
+        // Shut down the server, simulating the scenario of the process containing the server terminating
+        (server as! XPCAnonymousServer).shutdown()
+        
+        // Make two more calls after having shutdown the server:
+        // - The second call should fail upon trying to send the message, connection is interrupted
+        // - The third call (and any subsequent ones) should fail indicating no new connections can ever be established
+        let interruptedExpectation = self.expectation(description: "Second message results in an interrupted error")
+        let cannotBeReestablishedExpectation = self.expectation(description: "Third message can't be sent")
+        try client.sendMessage("2nd message", route: echoRoute) { response in
+            switch response {
+                case .failure(let error):
+                    switch error {
+                        case .connectionInterrupted:
+                            interruptedExpectation.fulfill()
+                            
+                            // make another call to the server and this should fail with connectionCannotBeReestablished
+                            do {
+                                try client.sendMessage("3rd message", route: echoRoute) { _ in }
+                                XCTFail("No error was thrown. \(XPCError.connectionCannotBeReestablished) should " +
+                                        "have been thrown.")
+                            } catch XPCError.connectionCannotBeReestablished {
+                                cannotBeReestablishedExpectation.fulfill()
+                            } catch {
+                                XCTFail("Unexpected error: \(error). \(XPCError.connectionCannotBeReestablished) " +
+                                        "should have been thrown.")
+                            }
+                        default:
+                            XCTFail("Unexpected error: \(error). \(XPCError.connectionInterrupted) should have been " +
+                                    "returned.")
+                    }
+                case .success(_):
+                    XCTFail("No error was returned. \(XPCError.connectionInterrupted) should have been returned.")
```

Comment:
Swift's pattern matching is composable, so you can just do:

```suggestion
           switch response {
                case .failure(let .connectionInterrupted):
                    interruptedExpectation.fulfill()
                    
                    // make another call to the server and this should fail with connectionCannotBeReestablished
                    do {
                        try client.sendMessage("3rd message", route: echoRoute) { _ in }
                        XCTFail("No error was thrown. \(XPCError.connectionCannotBeReestablished) should " +
                                "have been thrown.")
                    } catch XPCError.connectionCannotBeReestablished {
                        cannotBeReestablishedExpectation.fulfill()
                    } catch {
                        XCTFail("Unexpected error: \(error). \(XPCError.connectionCannotBeReestablished) " +
                                "should have been thrown.")
                    }
                case .failure(let _):
                    XCTFail("Unexpected error: \(error). \(XPCError.connectionInterrupted) should have been returned.")
                case .success(_):
                    XCTFail("No error was returned. \(XPCError.connectionInterrupted) should have been returned.")
```

Reply:
Ah interesting, didn't realize that was possible. Incorporated this with a couple tweaks.